server/api/operator.go

```diff
@@ -125,177 +169,168 @@ func (h *operatorHandler) Post(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	name, ok := input["name"].(string)
+	name, ok := input[pName].(string)
 	if !ok {
-		h.r.JSON(w, http.StatusBadRequest, "missing operator name")
+		respWithMissParam(w, h.r, pName)
 		return
 	}
 
+	parseUInt64Param := func(name string) (uint64, bool) {
+		v, ok := input[name].(float64)
+		if !ok {
+			respWithMissParam(w, h.r, name)
+			return 0, ok
+		}
+		return uint64(v), ok
+	}
+
 	switch name {
-	case "transfer-leader":
-		regionID, ok := input["region_id"].(float64)
+	case opTransferLeader:
+		regionID, ok := parseUInt64Param(pRegionID)
 		if !ok {
-			h.r.JSON(w, http.StatusBadRequest, "missing region id")
 			return
 		}
-		storeID, ok := input["to_store_id"].(float64)
+
+		storeID, ok := parseUInt64Param(pToStoreID)
 		if !ok {
-			h.r.JSON(w, http.StatusBadRequest, "missing store id to transfer leader to")
 			return
 		}
-		if err := h.AddTransferLeaderOperator(uint64(regionID), uint64(storeID)); err != nil {
+		if err := h.AddTransferLeaderOperator(regionID, storeID); err != nil {
 			h.r.JSON(w, http.StatusInternalServerError, err.Error())
 			return
 		}
-	case "transfer-region":
-		regionID, ok := input["region_id"].(float64)
+	case opTransferRegion:
+		regionID, ok := parseUInt64Param(pRegionID)
```

Comment:
IMO, the original way is simple enough.

Reply:
here we use constant `pRegionID` instead of "region_id" so we needn't check whether the spelling of `region_id` is correct everywhere on those it used as the param for region's ID and we wish they should keep the same.

Reply:
I mean `input[pRegionID].(float64)`

Reply:
yes, the original way is very simple , the function `parseUIntParam` is used to parse and check the `UInt` param from `input` which only contains 3~4 lines and seems simple enough. However, the same logic of "parse and check the `Uint` param from `input`" appears serval times, so why we need the function `parseUInt64Param`:

- They allow us to conceive of our program as a bunch of sub-steps. (Each sub-step can be its own function. When any program seems too hard, just break the overall program into sub-steps!)

- They allow us to reuse code instead of rewriting it.